lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -49,11 +49,18 @@ def onNewMessage(ctx, message)
         @nocodec_transformer.transform(event)
         @queue << event
       else
-        codec(ctx).accept(CodecCallbackListener.new(target_field,
-                                                    hash,
-                                                    message.getIdentityStream(),
-                                                    @codec_transformer,
-                                                    @queue))
+        current_codec = codec(ctx)
+        if current_codec
```

Comment:
review note: this condition allows  `BeatsHandler#channelRead0()` to continue without crashing on `messageListener.onNewMessage(ctx, message);`. But since reproducing the issue is very hard (I couldn't with multiple tries), I have no idea what will happen next. So focusing on only solving the `(NoMethodError) undefined method 'accept' for nil:NilClass` error.

My concern is, since connection is not available, [ack operation (`BeatsHandler#writeAck`)](https://github.com/logstash-plugins/logstash-input-beats/blob/main/src/main/java/org/logstash/beats/BeatsHandler.java#L131) or `ctx.flush()` still would fail.

Reply:
> By better exception I mean, BeatsHandler#exceptionCaught fails to send super.exceptionCaught(ctx, cause); but gets was submitted.... handler didn't handle last exception cyclic message.

Please could you describe better this sentence, I don't grasp it. In particular under which condition `exceptionCaught` fails to invoke `super.exceptionCaught`. From actual code https://github.com/logstash-plugins/logstash-input-beats/blob/5ff649a7561f6e25a67b2c373ca265cd1a81a05d/src/main/java/org/logstash/beats/BeatsHandler.java#L88-L101 it could fail if:
- `isNoisyExecption()` fails itself
or
- the exception is not nosy (no connection reset by the peer) _and_
- `extractCause` fails itself

Reply:
It means any exception isn't handled by overridden `BeatsHandler#exceptionCaught` method causes internal netty exception and we see exception flood (exception content has `was submitted.... but didn't handle last exception`), usually raised in channels. `BeatsHandler#exceptionCaught` needs to either let netty know the exception by calling `super.exceptionCaught(ctx, cause);` or swallow it if expected behavior (`isNoisyException` or `RejectedExecutionException` for this case).